### ISSUE TITLE
Element 'css', attribute 'as': The attribute 'as' is not allowed. (CSS preloading)

### ISSUE THREAD

**Issue by benoitrenaud** [magento/magento2#18347]
Hi, 
We want to use a technique for loading asynchronous stylesheets. 
Ex. https://brianflove.com/2016/07/21/preloading-with-critical-path-css/

The issue is that link attribute "as" isn't valid according to vendor/magento/framework/View/Layout/etc/head.xsd validation file but is a officially supported HTML link attribute as described here: 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#Attributes

### Preconditions

1. Magento 2.2.6
2. PHP 7.1

### Steps to reproduce

1. I add the attribute **AS** on my css inclusion:

```xml
<?xml version="1.0"?>
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <head> 
        <css src="/style.css" src_type="url" rel="preload" as="style" /> 
    </head>
</page>
```
2. Clear cache
3. Reload the front

### Expected result

1. Permit the **AS** attribute (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#Attributes) in this file vendor/magento/framework/View/Layout/etc/head.xsd
2. Add this <xs:attribute name="as" type="xs:string"/> under <xs:complexType name="linkType"> 

### Actual result

An exception is thrown by Magento Framework 1 exception(s): Exception #0 (Magento\Framework\Config\Dom\ValidationException): Element 'css', attribute 'as': The attribute 'as' is not allowed.

**Comment by magento-engcom-team** [comment]
Hi @benoitrenaud. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@benoitrenaud do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @benoitrenaud thank you for you report, as devdocs says https://devdocs.magento.com/guides/v2.0/frontend-dev-guide/layouts/xml-instructions.html attribute AS may be supported.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96253`, `MAGETWO-96254` were created

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @benoitrenaud. Thank you for your report. 
The issue has been fixed in magento/magento2#20495 by @vasilii-b in 2.3-develop branch
Related commit(s):
 - [57f6957df12c1d9997f88e10f1262087e6d0a66a](https://github.com/magento/magento2/commit/57f6957df12c1d9997f88e10f1262087e6d0a66a)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @benoitrenaud. Thank you for your report. 
The issue has been fixed in magento/magento2#21261 by @amol2jcommerce in 2.2-develop branch
Related commit(s):
 - [46572afd1a9f142958207718311dad47b667d606](https://github.com/magento/magento2/commit/46572afd1a9f142958207718311dad47b667d606)

The fix will be available with the upcoming 2.2.9 release.
